File: Euler.py
## Problem 1
sum = 0

for x in range(1000):
    if x%3 == 0 or x%5 == 0:
        sum += x

print(sum)

## Problem 2
cur = 1
next = 1
evensum = 0

while next < 4000000:
    temp = cur + next
    cur = next
    next = temp

    if next%2 == 0:
        evensum += next

print(evensum)

## Problem 3
from sympy import *

target = 600851475413
factor = 2
largest = 0

while factor < target/2:
    if target%factor == 0:
        largest = factor
    factor = nextprime(factor)

print(largest)

File: Euler.R
# Problem 1
nums <- 1:1000
total = 0

for(x in nums) {
  if(x%%3 == 0 | x%%5 == 0){
    total = total + x
  }
}

print(total)

# Problem 2
cur = 1
new = 1
total2 = 0

while(new < 4000000){
  temp = cur + new
  cur = new
  new = temp
  
  if(new%%2 == 0){
    total2 = total2 + new
  }
}

print(total2)

# Problem 3
factors = primeFactors(600851475143)
print(factors[length(factors)])
